src/app/zap-templates/templates/app/cluster-commands-src.zapt

```diff
@@ -15,25 +15,7 @@ namespace {{asUpperCamelCase name}} {
 
 {{#if isFabricScoped~}}
 {{#if (isServer source)}}
-CHIP_ERROR Type::EncodeForRead(TLV::TLVWriter & aWriter, TLV::Tag aTag, FabricIndex aAccessingFabricIndex) const {
-    DataModel::WrappedStructEncoder encoder{aWriter, aTag};
-    {{#zcl_command_arguments~}}
-
-    {{~#if_is_fabric_scoped_struct type~}}
-    encoder.EncodeForRead(to_underlying(Fields::k{{asUpperCamelCase label}}), aAccessingFabricIndex, {{asLowerCamelCase label}});
```

Comment:
This was the other reason that the method name matching worked nicely, i.e. we had `EncodeForRead` all the way down

What if instead we change this for both structs & commands to be `EncodeForClient`?

Reply:
That's pretty vague in the sense that it's not clear which direction ForClient is in.

Let's try to get our use cases pinned down here and then we can see whether naming changes make sense.

We have the following:

1. Encode a list of fabric-scoped structs for attribute write.
2. Encode a list of fabric-scoped structs for attribute read.
3. Encode a fabric-scoped client-to-server command; when this has fabric-scoped structs (or lists of those) in the payload they should be encoded the same way as in item 1, right?
4. Encode a server-to-client command; when this has fabric-scoped structs (or lists of those) in the payload, they should be encoded the same way as in item 2.

Does that accurately describe the situation?

Reply:
All correct 👍